lighthouse-cli/test/smokehouse/run-lighthouse-cli.js

```diff
@@ -0,0 +1,130 @@
+/**
+ * @license Copyright 2019 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const fs = require('fs').promises;
+const path = require('path');
+const {promisify} = require('util');
+const execAsync = promisify(require('child_process').exec);
+
+const log = require('lighthouse-logger');
+const rimraf = promisify(require('rimraf'));
+const makeDir = require('make-dir');
+
+const assetSaver = require('../../../lighthouse-core/lib/asset-saver.js');
+const LocalConsole = require('./local-console.js');
+const ChildProcessError = require('./child-process-error.js');
+
+/**
+ * Launch Chrome and do a full Lighthouse run via the CLI.
+ * @param {string} url
+ * @param {LH.Config.Json=} configJson
+ * @param {{isDebug?: boolean}=} testRunnerOptions
+ * @return {Promise<{lhr: LH.Result, artifacts: LH.Artifacts, log: string}>}
+ */
+async function runLighthouse(url, configJson, testRunnerOptions = {}) {
+  const randInt = Math.round(Math.random() * 100000);
+  const tmpAbsolutePath = await makeDir(`./.tmp/smokehouse-${randInt}/`);
+  const tmpPath = path.relative(process.cwd(), tmpAbsolutePath);
+
+  const {isDebug} = testRunnerOptions;
+  return internalRun(url, tmpPath, configJson, isDebug)
+    // Wait for internalRun() before rimraffing scratch directory.
+    .finally(() => !isDebug && rimraf(tmpPath));
+}
+
+/**
+ * Internal runner.
+ * @param {string} url
+ * @param {string} tmpPath
+ * @param {LH.Config.Json=} configJson
+ * @param {boolean=} isDebug
+ * @return {Promise<{lhr: LH.Result, artifacts: LH.Artifacts, log: string}>}
+ */
+async function internalRun(url, tmpPath, configJson, isDebug) {
+  const localConsole = new LocalConsole();
+
+  const outputPath = `${tmpPath}/smokehouse.report.json`;
+  const artifactsDirectory = `${tmpPath}/artifacts/`;
+
+  const args = [
+    'node',
+    'lighthouse-cli/index.js',
+    `"${url}"`, // quoted for weird urls
```

Comment:
consider using `execFileAsync`, so it will handle all the arg escaping for us

Reply:
> consider using `execFileAsync`, so it will handle all the arg escaping for us
> 
> ex: the `-G=${artifactsDirectory}` part breaks if the cwd has a space in it.

arg, arguments.

Thanks, I'll try it out.

Reply:
resolved but the code isn't changed - did you forget to push?

Reply:
> resolved but the code isn't changed - did you forget to push?

still working, still working :)